lib/connect/ws.js

```diff
@@ -2,14 +2,36 @@
 
 var websocket = require('websocket-stream')
 var _URL = require('url')
-var wssProperties = [
-  'rejectUnauthorized',
-  'ca',
-  'cert',
-  'key',
-  'pfx',
-  'passphrase'
-]
+var mqttProperties = [
+'keepalive',
+'reschedulePings',
+'clientId',
+'protocolId',
+'protocolVersion',
+'clean',
+'reconnectPeriod',
+'connectTimeout',
+'username',
+'password',
+'incomingStore',
+'outgoingStore',
+'queueQoSZero',
+'will',
+];
+
+var urlKeys = [
+'href',
+'slashes',
+'path',
+'protocol',
+'auth',
+'hostname',
+'port',
+'host',
+'pathname',
+'search',
+'query',
+'hash' ];
```

Comment:
I'll try sorry.

the 3. way would be to add a Options key to the list of defined once:

> options is the client connection options (see: the connect packet). Defaults:
keepalive: 10 seconds, set to 0 to disable
reschedulePings: reschedule ping messages after sending packets (default true)
clientId: 'mqttjs_' + Math.random().toString(16).substr(2, 8)
protocolId: 'MQTT'
protocolVersion: 4
clean: true, set to false to receive QoS 1 and 2 messages while offline
reconnectPeriod: 1000 milliseconds, interval between two reconnections
connectTimeout: 30 * 1000 milliseconds, time to wait before a CONNACK is received
username: the username required by your broker, if any
password: the password required by your broker, if any
incomingStore: a Store for the incoming packets
outgoingStore: a Store for the outgoing packets
queueQoSZero: if connection is broken, queue outgoing QoS zero messages (default true)
will: a message that will sent by the broker automatically when the client disconnect badly. The format is:
topic: the topic to publish
payload: the message to publish
qos: the QoS
retain: the retain flag

because we only have one option object and need to handle options for mqtt and for websockets 

so we could add a key to the given list like 

> wsOptions: options for the websocket connection 

then we can simply handover this wsOptions object to the websocket constructor and don't need list options we want to support (way 1) or remove the overhead.(way 2)
the overhead is everything in the options object we need for mqtt but not for the websocket connection.